pkg/endpointmanager/manager.go

```diff
@@ -276,10 +251,33 @@ func TriggerPolicyUpdates(owner endpoint.Owner) *sync.WaitGroup {
 				} // else policy changed, but can't regenerate => do not change status
 			}
 			wg.Done()
-		}(Endpoints[k], &wg)
+		}(ep, &wg)
 	}
 
-	Mutex.RUnlock()
-
 	return &wg
 }
+
+// HasGlobalCT returns true if the endpoints have a global CT, false otherwise.
+func HasGlobalCT() bool {
+	eps := GetEndpoints()
+	for _, e := range eps {
+		e.RLock()
+		globalCT := e.Consumable != nil && !e.Opts.IsEnabled(endpoint.OptionConntrackLocal)
+		e.RUnlock()
+		if globalCT {
+			return true
+		}
+	}
+	return false
+}
+
+// GetEndpoints returns a slice of all endpoints present in endpoint manager.
+func GetEndpoints() []*endpoint.Endpoint {
+	mutex.RLock()
+	var eps []*endpoint.Endpoint
```

Comment:
I doubt it will ever really be an issue but we can `make([]*endpoint.Endpoint, len(endpoints))` to ensure append doesn't have to do any allocations.

Reply:
You meant `	eps := make([]*endpoint.Endpoint, 0, len(endpoints))` right?